maven-plugin-tests/basic-codegen-test/pom.xml

```diff
@@ -0,0 +1,127 @@
+                    <execution>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>generate</goal>
+                        </goals>
+                        <configuration>
+                            <projectSettings>
+                                <mavenFileStructure>false</mavenFileStructure>
+                                <includeSpec>false</includeSpec>
+                                <codeOnly>true</codeOnly>
+                                <reactive>false</reactive>
+                                <cliGenCI>false</cliGenCI>
+                                <groupId>io.apicurio.codegen.tests</groupId>
+                                <artifactId>basic-codegen-test</artifactId>
+                                <javaPackage>io.apicurio.codegen.tests.basic.api</javaPackage>
+                            </projectSettings>
+                            <inputSpec>openapi.json</inputSpec>
+                            <outputDir>${project.build.directory}/generated-sources/jaxrs/</outputDir>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>build-helper-maven-plugin</artifactId>
+                <version>3.3.0</version>
+                <executions>
+                    <execution>
+                        <id>add-source</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>add-source</goal>
+                        </goals>
+                        <configuration>
+                            <sources>
+                                <source>${project.build.directory}/generated-sources/jaxrs/</source>
+                            </sources>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+        <pluginManagement>
+            <plugins>
+                <!--This plugin's configuration is used to store Eclipse m2e settings only. It has no influence 
+                    on the Maven build itself. -->
+                <plugin>
+                    <groupId>org.eclipse.m2e</groupId>
+                    <artifactId>lifecycle-mapping</artifactId>
+                    <version>1.0.0</version>
+                    <configuration>
+                        <lifecycleMappingMetadata>
+                            <pluginExecutions>
+                                <pluginExecution>
+                                    <pluginExecutionFilter>
+                                        <groupId>io.apicurio</groupId>
+                                        <artifactId>apicurio-codegen-maven-plugin</artifactId>
+                                        <versionRange>[1.0.12-SNAPSHOT,)</versionRange>
```

Comment:
I think that we can use some "maven invoker" templating here, like:
https://github.com/fabric8io/kubernetes-client/blob/e520e0dd84b2a7b94d4f0bc8ee72a02bd056172d/java-generator/it/src/it/enum-ser-deser/pom.xml#L82

Reply:
Actually I didn't intend to check that in.  I'll remove it.